packages/react/react/src/hooks/useCombinedRefs.ts

```diff
@@ -18,17 +16,16 @@ type CallbackRef<T> = (ref: T | null) => void;
  *   return <div ref={ref} />;
  * })
  */
-export function useCombinedRefs<T>(...refs: Array<Ref<T> | CallbackRef<T>>): Ref<T> {
-  return useCallback(
-    (value: T) => {
-      for (const ref of refs) {
+export const useCombinedRefs = <T>(...refs: Array<Ref<T>>): RefCallback<T> =>
+  useCallback(
+    (value: T) =>
+      refs.forEach(ref => {
         if (typeof ref === 'function') {
           ref(value);
         } else if (ref != null) {
           (ref as MutableRefObject<T>).current = value;
         }
-      }
-    },
-    [refs]
+      }),
```

Comment:
I think using for-loops or Array#forEach is a matter of choice and style, and prefer to keep the original code unless there are some specific reasons. 

If they change a lot, there would be many minor changes in the pull request, making it hard to see the purpose of the pull request (because it contains a lot of unrelated changes). Also, in the long run, it will make it hard to track the changes and history of the code. What do you think?

Reply:
I agree with @raon0211 .

In the case of `useCombinedRefs`, I think there is no dramatic performance improvement between forEach and for loop. So if there is no specific reason to change it, we should keep for-loop.

Reply:
In addition, I don't know what the actual problem this PR is solving.

Reply:
First of all, I tried to run the hook in the test code as in the example, but an error occurred.

Second, CallbackRef is newly defined, which is already defined as RefCallback in the Ref type.

Am I thinking wrong?

Reply:
And although you already solved it for me, I was able to find the same problem as [this](https://github.com/toss/slash/pull/212).

Reply:
This is not the type error because of `CallbackRef` or because we didn't allow undefined.

`useCombinedRefs` is a hook to update the passed refs with one same value.

But, in the screenshot you attached, the passed refs to `useCombinedRefs` are not accepting the same type of the value. Therefore, the type error should be occurred.

I'm sorry but I think your case is incorrect and the hook works well.

Reply:
To sum up, all passed refs should accept the same type. So they can be updated at the same time by `useCombinedRefs`.

Reply:
Oh, I see, I think I'm misunderstanding. Thanks for telling me. Another story, but I have an additional question.

I was using type assertion but I took refObject readonly and changed it to MutableRefObject. I had only bad thoughts about type assertions, but this code changed my mind. am i right?

If so, it's so cool and I wonder why you're doing it this way. Because we know that relying on type inference and type checking will make your code safer and more maintainable in the long run. We also know that it helps us detect potential issues early in the development process. So I wonder why they made this change!

Reply:
Also, I used the example stated in the jsdoc, not the example I used. If so, is it correct to change the jsdoc correctly?

Reply:
thank you @HoseungJang 

I'm here because I want to contribute, but I'm just learning and going. So what should I do with this PR?

Reply:
I think why not close this pull request, and go on to the next task? 🚀

Reply:
good @raon0211

Reply:
My challenge for SLASH continues!!